### ISSUE TITLE
First build attempt - failed!

### ISSUE THREAD

**Issue by brucou** [pikapkg/pack#39]
Hello there,

I want to report that, after raising expectations with such nice promises, it is frustrating to fail at the first attempt to build anything.

So for those who pass through the same misery, I wanted to write here how I solved issues:

- Error messages was 

```
@pika/pack build v0.3.5
Error: Cannot find module '@pika/plugin-standard-pkg'
```

Speculated reason: while the README says to install @pika with the global `-g` option, it says nothing about the modules. I installed also through `npm install -g` and get this error.

Workaround: install the modules locally with `npm install --save` instead of globally

- next error was :  

```
Error: Multiple configuration files found. Please remove one:
 - .babelrc
 - .babelrc.js
```

here is the `.babelrc`:
```
{
  "presets": ["./.babelrc.js"]
}
```

So apparently that is a pattern you cannot use with pika, as it detects `.babelrc.js` also as a `.babelrc` file.

So deleted the `babelrc.js` file after copying it into `.babelrc`:

```
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "modules": false,
        "loose": true
      }
    ],
    "@babel/preset-react"
  ],
  "plugins": [
    "transform-class-properties",
    "transform-object-rest-spread",
    "babel-plugin-idx",
    "annotate-pure-calls",
    [
      "transform-react-remove-prop-types",
      {
        "mode": "unsafe-wrap"
      }
    ]
  ]
}
``` 

Error now is :   

```
@pika/pack build v0.3.5
[1/7] Validating source...
[2/7] Preparing pipeline...
      ❇️  pkg\
[3/7] Running @pika/plugin-standard-pkg...
Error: Cannot find module '@babel/preset-react' from (...)
```

After subsequently fighting with babel, its dependencies, deleting all plugins, and upgrading to version 7, yay!

```
@pika/pack build v0.3.5
[1/7] Validating source...
[2/7] Preparing pipeline...
      ❇️  pkg\
[3/7] Running @pika/plugin-standard-pkg...
      📝  pkg\dist-src\index.js [esnext]
      » Linting with standard-pkg...
[4/7] Running @pika/plugin-build-node...
      📝  pkg\dist-node\index.js [main]
[5/7] Running @pika/plugin-build-web...
      📝  pkg\dist-web\index.js [module]
[6/7] Running @pika/plugin-build-types...
      » no type definitions found, auto-generating...
      📝  pkg\dist-types\index.d.ts [types]
[7/7] Finalizing package...
      » copying LICENSE...
      » copying README.md...
      📝  pkg\package.json
      📦  pkg\
Done in 3.33s.
```

My previous rollup build now fails but pika seems to work. So I guess I lost something, and I won something. 

Here is the type file : 
```
export const COMMAND_RENDER: string;
export class Machine {
constructor(props: any);
componentDidMount(): void;
componentWillUnmount(): void;
forceUpdate(callback: any): void;
render(): any;
setState(partialState: any,callback: any): void;
}
export function NO_ACTIONS(): any;
export const NO_STATE_UPDATE: any[];
export function destructureEvent(eventStruct: any): any;
export function getStateTransducerRxAdapter(RxApi: any): any;
export function renderAction(params: any): any;
export function testMachineComponent(testAPI: any,testScenario: any,machineDef: any): void;
```

So my question is : how does the type file generation works? Is it meant to be a skeleton which the developer needs to fill in the void (e.g. any)?

Bottom line is the idea is good, but documentation including troubleshooting needs to be improved to a whole other level.

**Comment by brucou** [comment]
Unfortunately pika publish failed too :  

```
? Select semver increment or specify new version patch  0.8.1
? Will bump from 0.8.0 to 0.8.1. Continue? Yes

[1/7] Prerequisite checks...

(and it blocked there for 10mn, and the terminal cannot be killed through ^kill)
```

I don't mean that as a criticism but more like to tell you that there is some way to go. When publishing the produced `pkg` build with `npm publish` and testing the new build out in some codesandbox, it works fine. I also had to move pika dependencies such as `@pika/plugin-build-node`  from the `dependencies` property of `package.json` to `devDependencies` instead.

**Comment by brucou** [comment]
unfortunately this (pack publish) is still failing with the latest distribution of git and pika/pkg on windows. It is building correctly, but no luck for publishing. The error message is :

> × Unclean working tree. Commit or stash changes first.
> Publish failed. Rolling back to the previous state…

However, running `git status`:

```
nothing added to commit but untracked files present (use "git add" to track)
``
